### PR TITLE
Release v3.8.6

### PR DESCRIPTION
**Pull request by Dreamsorcerer**

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -10,6 +10,78 @@
...
+- Upgraded the vendored copy of llhttp_ to v9.1.3 -- by :user:`Dreamsorcerer`
+
+  Thanks to :user:`kenballus` for reporting this, see
+  https://github.com/aio-libs/aiohttp/security/advisories/GHSA-pjjw-qhg8-p2p9.
+
+  .. _llhttp: https://llhttp.org
+
+  `#7647 <https://github.com/aio-libs/aiohttp/issues/7647>`_
+
+- Updated Python parser to comply with RFCs 9110/9112 -- by :user:`Dreamorcerer`
```
Comment (webknjaz) [inline CHANGES.rst:28]: Nit: Sphinx has an `rfc` role for linking such stuff.

```suggestion
- Updated Python parser to comply with :rfc:`9110`/:rfc:`9112` -- by :user:`Dreamorcerer`
```

File: CHANGES.rst [hunk 2]
```diff
@@ -10,6 +10,78 @@
...
+
+
+Deprecation
+-----------
+
+- Added `fallback_charset_resolver` parameter in `ClientSession` to allow a user-supplied
+  character set detection function.
+
+  Character set detection will no longer be included in 3.9 as a default. If this feature is needed,
+  please use `fallback_charset_resolver <https://docs.aiohttp.org/en/stable/client_advanced.html#character-set-detection>`_.
```
Comment (webknjaz) [inline CHANGES.rst:43]: Can this be linked using the `ref` role?

File: CHANGES.rst [hunk 3]
```diff
@@ -10,6 +10,78 @@
...
+  Thanks to :user:`kenballus` for reporting this, see
+  https://github.com/aio-libs/aiohttp/security/advisories/GHSA-gfw2-4jvh-wgfg.
+
+  `#7663 <https://github.com/aio-libs/aiohttp/issues/7663>`_
+
+
+Deprecation
+-----------
+
+- Added `fallback_charset_resolver` parameter in `ClientSession` to allow a user-supplied
```
Comment (webknjaz) [inline CHANGES.rst]: This incorrectly uses single backticks.

File: CHANGES.rst [hunk 4]
```diff
@@ -10,6 +10,78 @@
...
+
+- Fixed ``PermissionError`` when .netrc is unreadable due to permissions.
+
+  `#7237 <https://github.com/aio-libs/aiohttp/issues/7237>`_
+
+- Fixed output of parsing errors pointing to a ``\n``. -- by :user:`Dreamsorcerer`
+
+  `#7468 <https://github.com/aio-libs/aiohttp/issues/7468>`_
+
+- Fixed GunicornWebWorker max_requests_jitter not working.
```
Comment (webknjaz) [inline CHANGES.rst]: Could use some highlighting.

File: CHANGES.rst [hunk 5]
```diff
@@ -10,6 +10,78 @@
...
+  (this should resolve some regressions when dealing with badly formatted HTTP responses). -- by :user:`Dreamsorcerer`
+
+  `#7490 <https://github.com/aio-libs/aiohttp/issues/7490>`_
+
+
+
+Bugfixes
+--------
+
+- Fixed ``PermissionError`` when .netrc is unreadable due to permissions.
```
Comment (webknjaz) [inline CHANGES.rst]: Use the `file` RST role?

File: CHANGES.rst [hunk 6]
```diff
@@ -10,6 +10,78 @@
...
+
+- Fixed output of parsing errors pointing to a ``\n``. -- by :user:`Dreamsorcerer`
+
+  `#7468 <https://github.com/aio-libs/aiohttp/issues/7468>`_
+
+- Fixed GunicornWebWorker max_requests_jitter not working.
+
+  `#7518 <https://github.com/aio-libs/aiohttp/issues/7518>`_
+
+- Fixed sorting in filter_cookies to use cookie with longest path. -- by :user:`marq24`.
```
Comment (webknjaz) [inline CHANGES.rst]: Highlight?

**Review by webknjaz — APPROVED** [review]
Minor things, but could be fixed.

File: CHANGES.rst (hunk 2 above)
Comment (Dreamsorcerer) [reply CHANGES.rst:43]: Not sure it'd work outside of Sphinx? e.g. when appearing in Github changelog.

Comment (Dreamsorcerer) [reply CHANGES.rst:43]: In fact, if you look at changelog for 3.8.2, I think several of your suggestions actually mess up rendering:

image

File: CHANGES.rst (hunk 1 above)
Comment (Dreamsorcerer) [reply CHANGES.rst:28]: Reverting this as per the other comment.

File: CHANGES.rst (hunk 2 above)
Comment (Dreamsorcerer) [reply CHANGES.rst:43]: I've updated the ones that seems like they'll render correctly, and ignored the other suggestions.

Comment (Dreamsorcerer) [reply CHANGES.rst:43]: On a similar point, maybe we use stop using the user role, as that doesn't render correctly either.

Comment (webknjaz) [reply CHANGES.rst:43]: I decided to ignore that long ago. You could add some post-processing but it's probably not worth it.

**Pull request merged**